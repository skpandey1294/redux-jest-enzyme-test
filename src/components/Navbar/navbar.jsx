import React from 'react';

import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';

import MenuIcon from '@material-ui/icons/Menu';

import { Link } from 'react-router-dom';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
  link: {
    color: '#fff',
    fontWeight: '1000',
    textDecoration: 'none'
  }
}));

function Navbar() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
            <MenuIcon />
          </IconButton>
          
          <Button color="inherit" className="user-btn">
          <Link to="/users" className={classes.link}>Users
          </Link>
          </Button>
          
          <Button color="inherit">
          <Link to="/"  className={classes.link}>
          Posts
          </Link>
          </Button>

          <Button color="inherit">
          <Link to="/user/albums" className={classes.link}>
          Albums
          </Link>
          </Button>
          
        </Toolbar>
      </AppBar>
    </div>
  );
}



export default Navbar