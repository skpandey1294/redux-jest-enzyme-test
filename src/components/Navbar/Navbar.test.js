import React from 'react'
import Navbar from './navbar'
import { shallow } from 'enzyme'


const setUp = (props = {}) => {
    const component = shallow(<Navbar />)
    return component
}

const findByTestAttr = (component, attr) => {
    const wrapper = component.find(`.${attr}`)
    return wrapper
}




describe('Testing the Navbar Component', () => {

    let component

    beforeEach(() => {
        component = setUp()
    })

    test('Navbar Test', () => { 
        // console.log(component.debug())
    })
    
    test('User button testing', () => {
        const wrapper = findByTestAttr(component, 'user-btn')
        expect(wrapper.length).toBe(1)
    })

    test('It should not fail', () => {
        const wrapper = findByTestAttr(component, 'user-btn')
        
        const element = findByTestAttr(wrapper, 'makeStyles-link-4')

        // console.log(element.debug())
        expect(element.length).toEqual(1)
    })
})
