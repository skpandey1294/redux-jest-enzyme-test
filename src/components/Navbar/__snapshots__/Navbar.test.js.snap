// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Navbar Component It should not fail 1`] = `ShallowWrapper {}`;
